torch/_inductor/jit_builder/cpp_builder.py

```diff
@@ -0,0 +1,1022 @@
+
+        cuda_definations: List[str] = []
+        cuda_include_dirs: List[str] = []
+        cuda_cflags: List[str] = []
+        cuda_ldflags: List[str] = []
+        cuda_libraries_dirs: List[str] = []
+        cuda_libraries: List[str] = []
+        cuda_passthough_args: List[str] = []
+
+        if use_cuda:
+            (
+                cuda_definations,
+                cuda_include_dirs,
+                cuda_cflags,
+                cuda_ldflags,
+                cuda_libraries_dirs,
+                cuda_libraries,
+                cuda_passthough_args,
+            ) = get_cpp_torch_cuda_options(aot_mode=aot_mode)
+
+        _append_list(self._definations, cuda_definations)
+        _append_list(self._include_dirs, cuda_include_dirs)
+        _append_list(self._cflags, cuda_cflags)
+        _append_list(self._ldflags, cuda_ldflags)
+        _append_list(self._libraries_dirs, cuda_libraries_dirs)
+        _append_list(self._libraries, cuda_libraries)
+        _append_list(self._passthough_args, cuda_passthough_args)
+        self._remove_duplicate_options()
+
+
+class CppBuilder:
+    """
+    CppBuilder is a cpp jit builder, and it supports both Windows, Linux and MacOS.
+    Args:
+        name:
+            1. Build target name, the final target file will append extension type automatically.
+            2. Due to the CppBuilder is supports mutliple OS, it will maintains ext for OS difference.
+
+        sources:
+            Source code file list to be built.
+
+        BuildOption:
+            Build options to the builder.
+
+        output_dir:
+            1. The output_dir the taget file will output to.
+            2. The default value is empty string, and then the use current dir as output dir.
+            3. Final target file: output_dir/name.ext
+    """
+
+    def get_shared_lib_ext(self) -> str:
+        SHARED_LIB_EXT = ".dll" if _IS_WINDOWS else ".so"
+        return SHARED_LIB_EXT
+
+    def get_object_ext(self) -> str:
+        EXT = ".obj" if _IS_WINDOWS else ".o"
+        return EXT
+
+    def __init__(
+        self,
+        name: str,
+        sources: List[str],
+        BuildOption: BuildOptionsBase,
+        output_dir: str = "",
+        compile_only: bool = False,
+        use_absolute_path: bool = False,
+    ) -> None:
+        self._compiler = ""
+        self._cflags_args = ""
+        self._definations_args = ""
+        self._include_dirs_args = ""
+        self._ldflags_args = ""
+        self._libraries_dirs_args = ""
+        self._libraries_args = ""
+        self._passthough_parameters_args = ""
+
+        self._output_dir = ""
+        self._target_file = ""
+
+        self._name = name
+
+        if config.is_fbcode():
+            if BuildOption.get_aot_mode() and use_absolute_path:
+                # We need to copy any absolute-path torch includes
+                inp_name = [os.path.basename(i) for i in sources]
+                self._sources_args = " ".join(inp_name)
```

Comment:
Missing `self._sources_args` in the else branch

Reply:
Tried to fixed.